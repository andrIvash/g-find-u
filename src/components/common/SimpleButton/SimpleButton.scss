@import '../../../styles/abstracts/functions';

$button-min-width: rem(140);
$button-radius: rem(2);

.simpleButton {
    button {
        min-width: $button-min-width;
        border-radius: $button-radius;
    }
}