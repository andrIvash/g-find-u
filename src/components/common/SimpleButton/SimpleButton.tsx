import * as React from 'react';
import Button from '@mui/material/Button';
import { Props , defaultLabel } from "./index";

import './SimpleButton.scss';

export const SimpleButton = (props: Props) => {
    
    const handleClick = () => {
        props.onClick();
    };

    return (
        <div className='simpleButton' data-testid="SimpleButton">
             <Button variant="contained" onClick={handleClick}>
                {props.label ? props.label : defaultLabel}
            </Button>
        </div>
    )
}
