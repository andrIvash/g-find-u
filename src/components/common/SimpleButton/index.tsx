import { SimpleButton } from './SimpleButton';

export const defaultLabel = "Ok";

export interface Props {
    label?: string,
    onClick: () => void,
}

export default SimpleButton;
