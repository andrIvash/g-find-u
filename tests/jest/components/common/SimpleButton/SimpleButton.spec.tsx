import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import SimpleButton from '../../../../../src/components/common/SimpleButton';

describe('SimpleButton Component', () => {
    const mockOnClick = jest.fn();
  
    it('renders correctly with provided label', () => {
      const label = 'Click Me';
      render(<SimpleButton label={label} onClick={mockOnClick} />);
      expect(screen.getByText(label)).toBeInTheDocument();
    });

    it('renders correctly with default label', () => {
        const defaultLabel = 'Ok';
        render(<SimpleButton onClick={mockOnClick} />);
        expect(screen.getByText(defaultLabel)).toBeInTheDocument();
      });
  
    it('calls onClick prop when clicked', () => {
      const label = 'Click Me';
      render(<SimpleButton label={label} onClick={mockOnClick} />);
      fireEvent.click(screen.getByText(label));
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  
    it('matches snapshot', () => {
      const label = 'Click Me';
      const { asFragment } = render(<SimpleButton label={label} onClick={mockOnClick} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });