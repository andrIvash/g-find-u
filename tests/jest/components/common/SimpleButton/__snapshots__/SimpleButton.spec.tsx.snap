// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="simpleButton"
    data-testid="SimpleButton"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Click Me
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
